File: wordCloud.js
function wordCloud(sentence) {

  sentence = sentence.replace(/[\.,-\/#!?$%\^&\*;:{}=\-_`~()]/g,"");

  sentence = sentence.replace(/\s{2,}/g,"");

  var newSentence = sentence.toLowerCase().split(" ");

  var word_count = {};

  for (var i = 0; i < newSentence.length; i++) {
    word_count[newSentence[i]] = 0
  };

  for (var i = 0; i < newSentence.length; i++) {
    word_count[newSentence[i]] += 1
  };

  console.log(word_count);
}

wordCloud("Hello. Hello. How are are you you, today?");

// this code is really, really ugly.

File: word_cloud.rb
# scenerios: sentence with puncuation, gsub with space
# then split the puncuation into a downcase array
# create a new hash
# iterate over each word and add 1 to each character based
# on how many times it shows up


def word_cloud(sentence)

  puncuation = %w(, . / ( ) ' ' " " ; !)

  puncuation.each {|punc| sentence = sentence.gsub(punc,"")}

  sentence = sentence.downcase.split(" ")

  word_count = Hash.new(0)
  sentence.each do |word|
    word_count[word] += 1
  end

  p word_count

end

word_cloud "Hello. Hello. What a fine day, today."